frontend/src/bundles/common/components/slider/slider.tsx

```diff
@@ -0,0 +1,60 @@
+import { Box, Slider } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { defaultExperienceMarks } from './default-values.js';
+import classes from './style.module.css';
+import { defaultSliderStyle } from './styles.js';
+
+type Mark = {
+    label?: string;
+    value: number;
+};
+
+type Properties = {
+    label?: string;
+    value?: number;
+    marks?: Mark[] | boolean | undefined;
+    className?: string;
+};
+
+const CustomSlider: React.FC<Properties> = ({
+    label = 'slider label',
+    value,
+    marks = defaultExperienceMarks,
+    className = classes.sliderContainerStyle,
+}) => {
+    const StyledSlider = styled(Slider)(defaultSliderStyle);
+
+    const getValueLabel = useCallback(
+        (value: number): string | undefined => {
+            return Array.isArray(marks)
+                ? (marks.find((mark) => mark.value === value)?.label as string)
+                : 'undefined';
```

Comment:
First of all your function now returns 'undefined' - string instead of undefined, if you want to return undefined.
Also returning undefined is bad practice, you can return null instead of undefined

```suggestion
        (value: number): string | null => {
            return Array.isArray(marks)
                ? (marks.find((mark) => mark.value === value)?.label as string)
                : null;
```

Reply:
agree, I will make appropriate changeы